session.go

```diff
@@ -398,10 +420,19 @@ func Connect(ctx context.Context, opts ...ConnectOption) (Session, error) {
 		remoteUpdateErr = &empty
 	}
 
+	cfg.ClientTypes = append([]string{"library/official/go"}, cfg.ClientTypes...)
+	cfg.ClientVersions = append([]string{libraryAgentVersion}, cfg.ClientVersions...)
+
+	userAgent := fmt.Sprintf("%s %s",
+		strings.Join(cfg.ClientTypes, ","),
+		strings.Join(cfg.ClientVersions, ","),
+	)
+
 	auth := proto.AuthExtra{
 		Version:            libraryAgentVersion,
```

Comment:
Shouldn't this now be:

```go
Version: strings.Join(cfg.ClientVersions, ","),
....
UserAgent: strings.Join(cfg.ClientTypes, ","),
```

If I'm reading the rust code right, something like `child_client("foo", "1.0.0").child_client("bar", "2.0.0")` results in `UserAgent: "foo, bar", Version: "1.0.0,2.0.0"`, and I guess we have to zip those two lists together to correlated later.

If we want to mimic that, I think we want them split here too, right?

Reply:
apparently i cannot read, but also that's an interesting departure from the way we've used User-Agent up till now; it's always had a version string before. Both?